File: frontend/lib/data/model/general.ts
import { DM_AbsPrompt, DM_AbsUtterance } from "./abs";

export type GENERAL_TASK_T = "general";
export const GENERAL_TASK: GENERAL_TASK_T = "general";

type GeneralUtterance = {
    utt: string
}

export type DM_GeneralPrompt = DM_AbsPrompt<GENERAL_TASK_T> & GeneralUtterance

export type DM_GeneralUtterance = DM_AbsUtterance<GENERAL_TASK_T> & GeneralUtterance

File: backend/otgpt_hft/data_model/general.py
from typing import Literal

from .abs import DM_AbsPrompt, DM_AbsUtterance

GENERAL_TASK_T = Literal["general"]
GENERAL_TASK: GENERAL_TASK_T = "general"


class DM_GeneralPrompt(DM_AbsPrompt[GENERAL_TASK_T]):
    task: GENERAL_TASK_T = GENERAL_TASK
    utt: str

    def get_utt(self) -> str:
        return self.utt


class DM_GeneralUtterance(DM_AbsUtterance[GENERAL_TASK_T]):
    task: GENERAL_TASK_T = GENERAL_TASK
    utt: str

    def get_utt(self) -> str:
        return self.utt
